From da435877e057ed18d8a053685bbee1c85abea7fc Mon Sep 17 00:00:00 2001
From: seb <sebastien.royen@armadeus.com>
Date: Thu, 28 Mar 2013 09:58:54 +0100
Subject: [PATCH 1/1] Input: mma8450 - fix signed 12bits to 32bits conversion

Event value is wrong. Should be in range -2048 to 2047, but is in range 0 to 4095.
Use s8 to int conversion and remove 0xfff mask.

Signed-off-by: Sebastien Royen <sebastien.royen@armadeus.com>
---
 drivers/input/misc/mma8450.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/input/misc/mma8450.c b/drivers/input/misc/mma8450.c
index 480557f..f330969 100644
--- a/drivers/input/misc/mma8450.c
+++ b/drivers/input/misc/mma8450.c
@@ -123,9 +123,9 @@ static void mma8450_poll(struct input_polled_dev *dev)
 	if (ret < 0)
 		return;
 
-	x = ((buf[1] << 4) & 0xff0) | (buf[0] & 0xf);
-	y = ((buf[3] << 4) & 0xff0) | (buf[2] & 0xf);
-	z = ((buf[5] << 4) & 0xff0) | (buf[4] & 0xf);
+	x = ((int)(s8)buf[1] << 4) | (buf[0] & 0xf);
+	y = ((int)(s8)buf[3] << 4) | (buf[2] & 0xf);
+	z = ((int)(s8)buf[5] << 4) | (buf[4] & 0xf);
 
 	input_report_abs(dev->input, ABS_X, x);
 	input_report_abs(dev->input, ABS_Y, y);
-- 
1.7.9.5

